rust/crd/src/ldap.rs

```diff
@@ -0,0 +1,195 @@
+use stackable_operator::commons::ldap::LdapAuthenticationProvider;
+
+use crate::authentication::ResolvedAuthenticationClasses;
+
+#[derive(Clone, Debug)]
+pub struct DruidLdapSettings {
+    pub ldap: LdapAuthenticationProvider,
+}
+
+impl DruidLdapSettings {
+    pub fn new_from(
+        resolved_authentication_config: &ResolvedAuthenticationClasses,
+    ) -> Option<DruidLdapSettings> {
+        if let Some(authentication_class) =
+            resolved_authentication_config.get_ldap_authentication_class()
+        {
+            if let AuthenticationClassProvider::Ldap(ref provider) =
+                authentication_class.spec.provider
+            {
+                return Some(DruidLdapSettings {
+                    ldap: provider.clone(),
+                });
+            }
+        }
+        None
+    }
+
+    fn add_druid_system_authenticator_config(&self, config: &mut BTreeMap<String, Option<String>>) {
+        const PREFIX: &str = "druid.auth.authenticator.DruidSystemAuthenticator";
+
+        config.insert(format!("{PREFIX}.type"), Some("basic".to_string()));
+        config.insert(
+            format!("{PREFIX}.credentialsValidator.type"),
+            Some("metadata".to_string()),
+        );
+
+        // this line is left out, as we don't want to create an admin user
+        // # druid.auth.authenticator.DruidSystemAuthenticator.initialAdminPassword: XXX
+
+        config.insert(
+            format!("{PREFIX}.initialInternalClientPassword"),
+            Some("xxx_druid_system_internal_client_password_xxx".to_string()),
+        );
+        config.insert(
+            format!("{PREFIX}.authorizerName"),
+            Some("DruidSystemAuthorizer".to_string()),
+        );
+        config.insert(format!("{PREFIX}.skipOnFailure"), Some("true".to_string()));
+    }
+
+    fn add_ldap_authenticator_config(&self, config: &mut BTreeMap<String, Option<String>>) {
+        const PREFIX: &str = "druid.auth.authenticator.Ldap";
+
+        config.insert(format!("{PREFIX}.type"), Some("basic".to_string()));
+        config.insert(
+            format!("{PREFIX}.enableCacheNotifications"),
+            Some("true".to_string()),
+        );
+        config.insert(
+            format!("{PREFIX}.credentialsValidator.type"),
+            Some("ldap".to_string()),
+        );
+        config.insert(
+            format!("{PREFIX}.credentialsValidator.url"),
+            Some(self.credentials_validator_url()),
+        );
+        config.insert(
+            format!("{PREFIX}.credentialsValidator.bindUser"),
+            Some("xxx_ldap_bind_user_xxx".to_string()), // NOTE: this placeholder will be replaced from a mounted secret on container startup
+        );
+        config.insert(
+            format!("{PREFIX}.credentialsValidator.bindPassword"),
+            Some("xxx_ldap_bind_password_xxx".to_string()), // NOTE: this placeholder will be replaced from a mounted secret on container startup
+        );
```

Comment:
I would use constant for the placeholders. Same goes for `xxx_druid_system_internal_client_password_xxx`.